costools/add_cos_s_region.py

```diff
@@ -0,0 +1,107 @@
+#! /usr/bin/env python
+
+# Add the S_REGION keyword to raw files
+#
+import glob
+import argparse
+
+from calcos import cosutil
+from astropy.io import fits
+
+__version__ = '0.1'
+def main(dry_run=False):
+    """Add the S-region keyword to all COS FITS files in the current
+    directory that meet the following criteria:
+
+    Filename ends with one of ['rawtag.fits',
+                               'rawtag_a.fits',
+                               'rawtag_b.fits',
+                               'rawaccum.fits',
+                               'rawaccum_a.fits',
+                               'rawaccum_b.fits',
+                               'rawacq.fits']
+
+    The S_REGION keyword will have the value
+    'CIRCLE ra_aper dec_aper diameter'
+    where ra_aper and dec_aper are in decimal degrees and are read from
+    the extension 1 header, and diameter = 2.5/3600.0, which is the diameter
+    of the PSA and BOA in decimal degrees.
+
+    The keyword will be added after the PA_APER keyword in the extension 1
+    header
+
+    """
+    endings = ['rawtag.fits',
+               'rawtag_a.fits',
+               'rawtag_b.fits',
+               'rawaccum.fits',
+               'rawaccum_a.fits',
+               'rawaccum_b.fits',
+               'rawacq.fits']
+
+    fitsfiles = glob.glob('*.fits')
+    for file in fitsfiles:
+        for ending in endings:
+            if file.endswith(ending):
```

Comment:
Is it necessary to iterate over `endings`? Is it sufficient to check if the specific file ending is in `endings`?

Reply:
This seemed like the easiest way.  It's too fiddly extracting the ending from a file name, as they can have different lengths and numbers of underscores